File: js/state.js
export class MultisigWallet {
  signer1 = new Uint8Array(32);
  signer2 = new Uint8Array(32);
  signer3 = new Uint8Array(32);
  constructor(fields = undefined) {
    if (fields) {
      this.signer1 = fields.signer1 || this.signer1;
      this.signer2 = fields.signer2 || this.signer2;
      this.signer3 = fields.signer3 || this.signer3;
    }
  }
}

export const MultisigWalletSchema = new Map([
  [
    MultisigWallet,
    {
      kind: "struct",
      fields: [
        ["m", "u8"],
        ["is_initialized", "u8"],
        ["signer1", [32]],
        ["signer2", [32]],
        ["signer3", [32]],
      ],
    },
  ],
]);

export class MultisigWalletRequest {
  is_initialized = 0;
  is_signed1 = 0;
  is_signed2 = 0;
  is_signed3 = 0;
  wallet = new Uint8Array(32);
  constructor(fields = undefined) {
    if (fields) {
      this.amount = fields.amount;
      this.receiver = fields.receiver;
    }
  }
}

export const MultisigWalletRequestSchema = new Map([
  [
    MultisigWalletRequest,
    {
      kind: "struct",
      fields: [
        ["is_initialized", "u8"],
        ["amount", "u64"],
        ["wallet", [32]],
        ["receiver", [32]],
        ["is_signed1", "u8"],
        ["is_signed2", "u8"],
        ["is_signed3", "u8"],
      ],
    },
  ],
]);

File: src/state.rs
use borsh::{BorshDeserialize, BorshSerialize};
use solana_program::pubkey::Pubkey;

#[derive(BorshSerialize, BorshDeserialize, Debug)]
pub struct MultisigWallet {
    pub is_initialized: u8,
    pub m: u8,
    pub signer1: Pubkey,
    pub signer2: Pubkey,
    pub signer3: Pubkey,
}

#[derive(BorshSerialize, BorshDeserialize, Debug)]
pub struct Request {
    pub is_initialized: u8,
    pub is_finished: u8,
    pub amount: u64,
    pub wallet: Pubkey,
    pub receiver: Pubkey,
    pub is_signed1: u8,
    pub is_signed2: u8,
    pub is_signed3: u8,
}
